File: 0208-implement-trie-prefix-tree/0208-implement-trie-prefix-tree.cpp
class Trie {
public:
     vector<string>arr;   
    Trie() {
     
    }
    
    void insert(string word) {
        arr.push_back(word);
    }
    
    bool search(string word) {
        if(find(arr.begin(),arr.end(),word)!=arr.end())
            return true;
        return false;
    }
    
    bool startsWith(string prefix) {
        for(int i=0;i<arr.size();i++)
            if(arr[i].length()>=prefix.length())
            {
            if(arr[i].substr(0,prefix.length())==prefix)
                return true;
            }
        return false;
    }
};

/**
 * Your Trie object will be instantiated and called as such:
 * Trie* obj = new Trie();
 * obj->insert(word);
 * bool param_2 = obj->search(word);
 * bool param_3 = obj->startsWith(prefix);
 */

File: 0208-implement-trie-prefix-tree/0208-implement-trie-prefix-tree.java
class Trie {
   ArrayList<String>arr;
    public Trie() {
        arr=new ArrayList<>();
    }
    
    public void insert(String word) {
        arr.add(word);
    }
    
    public boolean search(String word) {
        return arr.contains(word);
    }
    
    public boolean startsWith(String prefix) {
        for(int i=0;i<arr.size();i++)
        {
            if(prefix.length()<=arr.get(i).length())
            if(arr.get(i).substring(0,prefix.length()).equals(prefix))
                return true;
        }
        return false;
    }
}

/**
 * Your Trie object will be instantiated and called as such:
 * Trie obj = new Trie();
 * obj.insert(word);
 * boolean param_2 = obj.search(word);
 * boolean param_3 = obj.startsWith(prefix);
 */